mteb/models/colbert_models.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Sequence
+from functools import partial
+from typing import Any
+
+import numpy as np
+import torch
+
+from mteb.encoder_interface import PromptType
+from mteb.model_meta import ModelMeta
+
+from .wrapper import Wrapper
+
+logger = logging.getLogger(__name__)
+
+
+class ColBERTWrapper(Wrapper):
+    def __init__(
+        self,
+        model_name: str,
+        **kwargs,
+    ) -> None:
+        """Wrapper for ColBERT models.
+
+        Args:
+            model_name: The ColBERT model to load from HuggingFace Hub.
+            **kwargs: Additional arguments to pass to the model.
+        """
+        try:
+            from pylate import models as colbert_model
+        except ModuleNotFoundError as e:
+            raise ModuleNotFoundError(
+                "To use the ColBERT models `pylate` is required. Please install it with `pip install mteb[pylate]`."
+            ) from e
+
+        self.model_name = model_name
+        self.static_model = colbert_model.ColBERT(self.model_name, **kwargs)
+
+    def encode(
+        self,
+        sentences: Sequence[str],
+        prompt_type: PromptType,
+        **kwargs: Any,
+    ) -> np.ndarray:
+        """Encodes the given sentences using the encoder.
+
+        Args:
+            sentences: The sentences to encode.
+            prompt_type: PromptType.query or PromptType.passage.
+            **kwargs: Additional arguments to pass to the encoder.
+
+        Returns:
+            The encoded sentences.
+        """
+        if "task_name" in kwargs:
+            kwargs.pop("task_name")
+        if "prompt_type" in kwargs:
+            kwargs.pop("prompt_type")
+
+        pred = self.static_model.encode(
+            sentences,
+            is_query=True if prompt_type == PromptType.query else False,
+            **kwargs,
+        )
+
+        pred = torch.nn.utils.rnn.pad_sequence(pred, batch_first=True, padding_value=0)
```

Comment:
Is this required? I couldn’t find any mention of it in the documentation. I think there should be an explanatory comment.